File: services/api/app/services/default_user.py
"""默认用户服务（单用户开发模式）"""
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select
from app.models.models import User

DEFAULT_ACCOUNT_ID = "default-user"


async def get_or_create_default_user(db: AsyncSession) -> User:
    result = await db.execute(
        select(User).where(User.account_id == DEFAULT_ACCOUNT_ID)
    )
    user = result.scalar_one_or_none()
    if not user:
        user = User(account_id=DEFAULT_ACCOUNT_ID, profile='{"name": "默认用户"}')
        db.add(user)
        await db.flush()
        await db.refresh(user)
    return user

File: apps/web/src/lib/default-user.ts
import { prisma } from "./prisma";

// 单用户模式下的默认账户 ID（无需登录系统）
const DEFAULT_ACCOUNT_ID = "default-user";

/**
 * 获取或创建默认用户
 * 开发/演示阶段使用单用户模式，无需认证
 */
export async function getOrCreateDefaultUser() {
  return prisma.user.upsert({
    where: { accountId: DEFAULT_ACCOUNT_ID },
    update: {},
    create: {
      accountId: DEFAULT_ACCOUNT_ID,
      profile: JSON.stringify({ name: "默认用户" }),
      preferences: "{}",
    },
  });
}
